sdk/tables/azure-data-tables/azure/data/tables/_authentication.py

```diff
@@ -32,6 +33,9 @@
     _wrap_exception,
 )
 
+if TYPE_CHECKING:
+    from typing import Union, Awaitable
```

Comment:
You can just import these from `typing` without needing to wrap it the `TYPE_CHECKING`.
You only need to do this for stuff imported from external libraries, like core, or other modules within this SDK.